### PR TITLE
RavenDB-16724: fixed - number of request incrementation. 

### PR DESCRIPTION
**Pull request by omerlewitz**
https://issues.hibernatingrhinos.com/issue/RavenDB-16724

### CODE REVIEW

**Review by ayende — APPROVED** [review]

**Comment by ppekrol** [comment]
@omerlewitz Can you check failing tests?

File: src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ public GetRevisionOperation(InMemoryDocumentSessionOperations session, IEnumerab
 
         public GetRevisionsCommand CreateRequest()
         {
+            _session.IncrementRequestCount();
```
Comment (karmeli87) [inline src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: I wonder if we don't want to have `IncrementRequestCount` in a more common place like in the `RequestExectuor.SendRequestToServer`, it is very easy to forget this increment.
For instance, does getting attachments increase the number of requests?

Comment (karmeli87) [reply src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: cc @ppekrol  thoughts?

Comment (omerlewitz) [reply src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: What about thread-safety if we change the implementation of the incrementation to `RequestExectuor.Increment..`

Comment (ayende) [reply src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: `RequestExectuor` isn't scoped to a session, it is global, so you cannot do that there.
We can't modify the API as well.

**Comment by omerlewitz** [comment]
> @omerlewitz Can you check failing tests?

Karmel suggested deleting the conditional statement of the `MaxNumbrOfRequests` because at `SlowTests.Cluster.ClusterTransactionTests.ClusterTransactionRequestWithRevisions` there's `while(true)` loop which breaks it.

File: src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: you do have the `SessionInfo`

Comment (ayende) [reply src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: Yes, that might work

Comment (ppekrol) [reply src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: That is a bigger change. Out of scope for this task. We can move it as a part of 5.3 effort.

Comment (omerlewitz) [reply src/Raven.Client/Documents/Session/Operations/GetRevisionOperation.cs:45]: ok :)

File: test/SlowTests/Issues/RavenDB-16724.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    Assert.Equal(3, session.Advanced.NumberOfRequests);
+                }   
+            }
+        }
+        private class User
+        {
+            public string Name { get; set; }
+        }
+        
+        public class AttachmentResult
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB-16724.cs]: You do have those classes available in Raven.Client, no need to copy them?

**Pull request merged**